File: 413-arithmetic-slices/413-arithmetic-slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        int n=nums.size();
        if(n<2) return 0;
        int diff=nums[1]-nums[0],st=0,ans=0;
        for(int i=1;i<n;i++){
            if(nums[i]-nums[i-1]!=diff){
                diff=nums[i]-nums[i-1];
                int cnt=i-st;
                st=i-1;
                ans+=((cnt-1)*(cnt-2))/2;
            }
        }
        int cnt=n-st;
        ans+=((cnt-1)*(cnt-2))/2;
        return ans;
    }
};

File: 413-arithmetic-slices/413-arithmetic-slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        ans,st,diff = 0,0,0
        for i in range(1,len(nums)):
            if nums[i]-nums[i-1]!=diff:
                diff = nums[i]-nums[i-1]
                cnt=i-st
                ans+=(cnt-1)*(cnt-2)//2
                st=i-1
        cnt=len(nums)-st
        ans+=(cnt-1)*(cnt-2)//2
        return ans